.github/workflows/main.yml

```diff
@@ -0,0 +1,27 @@
+name: build and test
+
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7', '3.8', '3.9', 'pypy-3.6']
```

Comment:
Python 3.6 end of security support is [2021-12-23](https://devguide.python.org/#status-of-python-branches). Why not drop it now? Similarly, we should switch to `pypy-3.7`.